client/app/scripts/superdesk-highlights/highlights.js

```diff
@@ -250,27 +250,67 @@
         };
     }
 
-    PackageHighlightsDropdownDirective.$inject = ['superdesk', 'desks', 'highlightsService'];
-    function PackageHighlightsDropdownDirective(superdesk, desks, highlightsService) {
+    PackageHighlightsDropdownDirective.$inject = ['superdesk', 'desks', 'highlightsService', '$location', '$route'];
+    function PackageHighlightsDropdownDirective(superdesk, desks, highlightsService, $location, $route) {
         return {
             templateUrl: 'scripts/superdesk-highlights/views/package_highlights_dropdown_directive.html',
             link: function(scope) {
-
-                scope.createHighlight = function(highlight) {
-                    highlightsService.createEmptyHighlight(highlight)
-                    .then(function(new_package) {
-                        superdesk.intent('author', 'package', new_package);
+                scope.$watch(function() {
+                    return desks.active;
```

Comment:
Listeners on a service are not automatically removed when the directive gets `$destroy`-ed, thus this `$watch` needs to be deregistered manually to prevent a memory leak.